prov/efa/src/efa_env.c

```diff
@@ -163,7 +162,14 @@ void efa_env_param_get(void)
 			    &efa_env.efa_read_segment_size);
 	fi_param_get_size_t(&efa_prov, "inter_max_gdrcopy_message_size",
 			    &efa_env.efa_max_gdrcopy_msg_size);
-	fi_param_get_bool(&efa_prov, "use_sm2", &efa_env.use_sm2);
+	fi_param_get_str(&efa_prov, "intranode_provider", &efa_env.intranode_provider);
+	fi_param_get_int(&efa_prov, "enable_shm_transfer", &efa_env.enable_shm_transfer);
```

Comment:
If both `intranode_provider` and `enable_shm_transfer` are set, then `enable_shm_transfer` will override `intranode_provider` just because it's parsed later. I think that's OK until we deprecate `enable_shm_transfer`.

Reply:
We are deprecating enable_shm_transfer now it will still override the behavior until we remove it from the code, and then it will do nothing.